cfpb/cfgov-refresh
# Add Data & Research app and Conference Registration form backend

Dada is implementing a form to register attendees of conferences for Research. This PR adds the `data_research` app and the modules necessary to stand up this form on a Browse page. For frontend review, this does change some templates around so I wanted you all to be aware.

## Additions

- `data_research` app to project
- ConferenceRegistrationForm, ConferenceRegistrationBlock, ConferenceRegistration model, ConferenceRegistrationHandler, and template
- data_research template root to template engine

## Changes

- Some atoms templates were changed to conform to Conference form needs

## Testing

- Run `cfgov/manage.py migrate`
- Run `tox`
- Create a Browse page and select Conference Registration Form block in the context Streamfield and add the fields. At least a gd code and a session.
- Publish the page.
- Visit the page and fill out the form
 - The first 3 fields are required and email must be properly formatted
- Submit the form
 - Note that the form should return an error because the GD code used is not registered with GD. This is the proper behavior for this circumstance.
 - Add a proper GD code to the page via the admin
 - Republish and resubmit the form. Notice the form should submit properly if all fields were filled out

Other things of note
- Failed submissions should have an error message and the values from the submissions filled out in the form.
- Successful submissions should have a success message and clear the  form

## Review

- @hillaryj 
- @cfpb/cfgov-backends 
- @jimmynotjim 
- @KimberlyMunoz 

## Screenshots
### Admin
![screen shot 2016-09-22 at 9 52 11 am](https://cloud.githubusercontent.com/assets/1412978/18750866/e8d50358-80aa-11e6-9a74-9a9236a97b8b.png)

### New Form
![screen shot 2016-09-22 at 9 49 40 am](https://cloud.githubusercontent.com/assets/1412978/18750874/f1c3f582-80aa-11e6-8403-864c4bd97a18.png)

### Bad Registration
![screen shot 2016-09-22 at 9 53 19 am](https://cloud.githubusercontent.com/assets/1412978/18750889/fb49b5ce-80aa-11e6-9af5-02d5787977ac.png)

### Bad form
![screen shot 2016-09-22 at 9 59 31 am](https://cloud.githubusercontent.com/assets/1412978/18750976/47312512-80ab-11e6-8fdb-d7f73f2d876b.png)

### Success
![screen shot 2016-09-22 at 9 55 26 am](https://cloud.githubusercontent.com/assets/1412978/18750894/025c9f2a-80ab-11e6-995e-bb6e92e33f98.png)


## Notes

- This is backend specific. No frontend validation is expected to work at this time. That's a todo below.

## Todos

- Add validation for checkbox selection
- Integrate frontend ajax request handling for form

## Checklist
